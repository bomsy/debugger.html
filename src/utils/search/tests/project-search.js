import searchSources, { searchSource } from "../project-search";
import fromJS from "../../fromJS";

const text = `
  function foo() {
    foo();
  }
`;

const text2 = `
  function bar() {
    foo();
  }
`;

const testFiles = fromJS({
  foo: {
    text,
    loading: false,
    id: "foo.js",
    url: "http://example.com/foo/foo.js"
  },
  foo2: {
    text: text2,
    loading: false,
    id: "foo.js",
    url: "http://example.com/foo/bar.js"
  }
});

describe("project search", () => {
  it("simple", () => {
    const source = {
      text,
      loading: false,
      id: "bar.js",
      url: "http://example.com/foo/bar.js"
    };

    const matches = searchSource(source, "foo");
    expect(matches).toMatchSnapshot();
  });

  it("searches multiple files", () => {
    const results = searchSources("foo", testFiles);
    const matches = results.map(i => i.matches);
    expect(results).toMatchSnapshot();
    expect(matches).toHaveLength(2);
    expect(matches[0]).toHaveLength(2);
    expect(matches[1]).toHaveLength(1);
  });

  it("handles no results", () => {
    const results = searchSources("fuzz", testFiles);
    const matches = results.map(i => i.matches);
    expect(results).toMatchSnapshot();
    expect(matches).toHaveLength(2);
    expect(matches[0]).toHaveLength(0);
    expect(matches[1]).toHaveLength(0);
  });
});
