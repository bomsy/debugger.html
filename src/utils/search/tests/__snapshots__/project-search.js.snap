// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project search handles no results 1`] = `
Array [
  Object {
    "filepath": "http://example.com/foo/foo.js",
    "matches": Array [],
    "source": Object {
      "id": "foo.js",
      "loading": false,
      "text": "
  function foo() {
    foo();
  }
",
      "url": "http://example.com/foo/foo.js",
    },
  },
  Object {
    "filepath": "http://example.com/foo/bar.js",
    "matches": Array [],
    "source": Object {
      "id": "foo.js",
      "loading": false,
      "text": "
  function bar() {
    foo();
  }
",
      "url": "http://example.com/foo/bar.js",
    },
  },
]
`;

exports[`project search searches multiple files 1`] = `
Array [
  Object {
    "filepath": "http://example.com/foo/foo.js",
    "matches": Array [
      Object {
        "column": 11,
        "line": 2,
        "match": "foo",
        "sourceId": "foo.js",
        "text": "  function foo() {",
        "value": "  function foo() {",
      },
      Object {
        "column": 4,
        "line": 3,
        "match": "foo",
        "sourceId": "foo.js",
        "text": "    foo();",
        "value": "    foo();",
      },
    ],
    "source": Object {
      "id": "foo.js",
      "loading": false,
      "text": "
  function foo() {
    foo();
  }
",
      "url": "http://example.com/foo/foo.js",
    },
  },
  Object {
    "filepath": "http://example.com/foo/bar.js",
    "matches": Array [
      Object {
        "column": 4,
        "line": 3,
        "match": "foo",
        "sourceId": "foo.js",
        "text": "    foo();",
        "value": "    foo();",
      },
    ],
    "source": Object {
      "id": "foo.js",
      "loading": false,
      "text": "
  function bar() {
    foo();
  }
",
      "url": "http://example.com/foo/bar.js",
    },
  },
]
`;

exports[`project search simple 1`] = `
Array [
  Object {
    "column": 11,
    "line": 2,
    "match": "foo",
    "sourceId": "bar.js",
    "text": "  function foo() {",
    "value": "  function foo() {",
  },
  Object {
    "column": 4,
    "line": 3,
    "match": "foo",
    "sourceId": "bar.js",
    "text": "    foo();",
    "value": "    foo();",
  },
]
`;
